docs/03_server/11_integration/07_data-pipeline/01_introduction.mdx

```diff
@@ -10,21 +10,90 @@ tags:
 
 Each Data Pipeline defines a source, some mapping operations and a sink.
 
 | term | description |
 | ---- | ----------- |
 | source | Changes in data at this location trigger your Data Pipeline |
-| map | Transforms data from the data source |
+| operators | Transform data from the data source |
 | sink | Does something with the mapped data from the data source |
 
+### Source
+Your pipeline needs a data source. The example below points to a local filesystem location. It pulls in any file that is dropped into the local file system’s directory **run/runtime/coppClarkSimple**:
+
+```kotlin
+source(
+    camelSource { 
+        location = "file://run/runtime/coppClarkSimple"
+    }
+)
+```
+
+### Operators
+Operators are used to transform input format to output format (although they can do more). You can chain them together, where each operator simply takes the output type of the previous one. 
+
+So in this case, the output of `.split` goes into `.map` and the output type of `.map` is a list of DbOperations, which the next sink (dbSync) can take as input.
+
+```kotlin
+            //1
+            .split(csvRawDecoder())
+            //2
+            .map { row ->
+                DbOperation.Upsert( ExchangeHoliday {
+                    //The exclamations at the end tell the code "we expect this to always be present in the row data"
+                    holidayDate = parseCCDateFormat(row.data["EventDate"])!!
+                    holidayName = row.data["EventName"]
+                    holidayType = getHolidayType(row.data["FileType"])!!
+                    isoMic = row.data["ISO MIC Code"]!!
+                })
+            }
+```
+
+The example above has two operators. Let's look at them more closely.
+
+- `split` uses the pre-built decoder which converts the CSV file and outputs a map containing each row in column + value: `CsvRow<Map<String, String>>`
+- `map` takes the data we need from the row columns and formats it accordingly. We are building up ExchangeHoliday, a database entity in our app, wrapped in the Db operation we want to perform on it; in this case, we want to upsert.
+
+For more complex requirements, you can create your own [transformer](#keeping-the-data-transformation-separate) in a seperate file. You can see this in two of our examples.
+
+Also, you can build multiple database entities from a single file. You can see details of this in the [third example](#updating-multiple-different-entities-from-a-single-source).
+
+### Sink
+Once the data is in the required format, we want to sink it. For file ingress, this means updating the database. 
+
+The example sink below takes the output from `.map` (a set of DB operations) and handles the updating of the database.
+
+```kotlin
+.sink(dbSink())
+```
+
+That’s all you need to ingest the data and put it in your database.
+
+However, there are additional helpers for things such as `onCompletion`. This enables you specify what happens after the file processing is complete. 
```

Comment:
I think there's a typo here: 'Enables you to specify...'

Reply:
Thanks, extra word inserted